micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/response/executor/RestExecutor.groovy

```diff
@@ -36,10 +39,17 @@ final class RestExecutor<T> {
         try {
             return exchangeInternal(params, httpMethod, responseType).get()
         } catch (ExecutionException e) {
-            throw e.cause
+            propagate(e.cause)
         }
     }
 
+    private void propagate(Throwable throwable) {
+        if (throwable instanceof TimeoutException) {
```

Comment:
Yes, we can just do Throwables.propagate, but IMHO wrapping TimeoutException allow you to handle exactly what you want without checking what is inside RuntimeException. Moreover I added Throwables.propagate only just in case because I don't know more checked exceptions except TimeoutException.